http.go

```diff
@@ -22,33 +22,57 @@ type (
 		UserAgent string
 		headers   *http.Header
 	}
+
+	// HTTPSourceClientOptFunc set some values to client
+	HTTPSourceClientOptFunc func(*HTTPSourceClient) error
```

Comment:
I don't see we are using the error return value with the current code... but are you suggesting that for extensibility?

Reply:
Yes, we will use someday... when add more options.